### ISSUE TITLE
RyuJIT/x86: Investigate SeekUnroll Failure

### ISSUE THREAD

**Issue by adiaaida** [dotnet/coreclr#8838]
In release x86 builds, we seem to have a silent bad codegen in SeekUnroll:

To repro:

build x86 release
<coreclr_root>bin\tests\Windows_NT.x86.Release\JIT\Performance\CodeQuality\SIMD\SeekUnroll\SeekUnroll\SeekUnroll.cmd

```
14:39:08 D:\j\workspace\x86_release_w---642e70e9\bin\tests\Windows_NT.x86.Release\TestWrappers\JIT\JIT.XUnitWrapper.cs(308535): error : JIT._Performance_CodeQuality_SIMD_SeekUnroll_SeekUnroll_SeekUnroll_._Performance_CodeQuality_SIMD_SeekUnroll_SeekUnroll_SeekUnroll_cmd [FAIL] [D:\j\workspace\x86_release_w---642e70e9\tests\runtest.proj]
14:39:08         cmdLine:D:\j\workspace\x86_release_w---642e70e9\bin\tests\Windows_NT.x86.Release\JIT\Performance\CodeQuality\SIMD\SeekUnroll\SeekUnroll\SeekUnroll.cmd Timed Out
14:39:08         
14:39:08   
14:39:08   Return code:      -100
14:39:08   Raw output file:      D:\j\workspace\x86_release_w---642e70e9\bin\tests\Windows_NT.x86.Release\Reports\JIT\Performance\CodeQuality\SIMD\SeekUnroll\SeekUnroll\SeekUnroll.output.txt
14:39:08   Raw output:
14:39:08   BEGIN EXECUTION
14:39:08          "D:\j\workspace\x86_release_w---642e70e9\tests\..\bin\tests\Windows_NT.x86.Release\Tests\Core_Root\corerun.exe" SeekUnroll.exe 
14:39:08         Index 1, times (ms) [50881, 50235, 50495, 51078, 50251, 50509, 49697, 49918, 50283, 48605]
14:39:08         Index 3, times (ms) [46499, 45958, 45131, 46220, 45834, 45438, 45447, 45670, 45735, 45703]
14:39:08         Index 11, times (ms) [83713, 82984, 83773, 83766, 86104, 83507, 84166, 83862, 83590, 83797]
14:39:08         cmdLine:D:\j\workspace\x86_release_w---642e70e9\bin\tests\Windows_NT.x86.Release\JIT\Performance\CodeQuality\SIMD\SeekUnroll\SeekUnroll\SeekUnroll.cmd Timed Out
14:39:08         Test Harness Exitcode is : -100
14:39:08         
14:39:08   To run the test:
14:39:08   > set CORE_ROOT=D:\j\workspace\x86_release_w---642e70e9\bin\tests\Windows_NT.x86.Release\Tests\Core_Root
14:39:08   > D:\j\workspace\x86_release_w---642e70e9\bin\tests\Windows_NT.x86.Release\JIT\Performance\CodeQuality\SIMD\SeekUnroll\SeekUnroll\SeekUnroll.cmd
14:39:08   
14:39:08         Expected: True
14:39:08         Actual:   False
14:39:08           D:\j\workspace\x86_release_w---642e70e9\bin\tests\Windows_NT.x86.Release\TestWrappers\JIT\JIT.XUnitWrapper.cs(308535,0): at JIT._Performance_CodeQuality_SIMD_SeekUnroll_SeekUnroll_SeekUnroll_._Performance_CodeQuality_SIMD_SeekUnroll_SeekUnroll_SeekUnroll_cmd()
```

**Comment by adiaaida** [comment]
/cc: @dotnet/jit-contrib 

This has been failing since (at least) Dec 16, which is the oldest run in the CI that we have.

**Comment by adiaaida** [comment]
I will try to track where this was introduced.

**Comment by adiaaida** [comment]
I sync'd back to a change on Dec 16, and this test passes, but is EXTREMELY slow. It is possible this is simply a timeout issue, not a silent bad codegen. If this passes at head of tree, we may need to modify the test or increase the timeout (significantly) to get this running on x86.

**Comment by adiaaida** [comment]
@JosephTremoulet It appears this test runs super slowly on x86 release builds (for example, it's been running for 40 minutes at this point and is only done with Index 11). For lab builds, this means that this test times out and makes it appear we are failing when really it's just a super long running test. Is there a way we can modify the test to make it run for less time? Should we just look into increasing the timeout, or should we disable this test for x86?

**Comment by JosephTremoulet** [comment]
Running time of this test is controlled by setting [InnerIterations] (https://github.com/dotnet/coreclr/blob/master/tests/src/JIT/Performance/CodeQuality/SIMD/SeekUnroll/SeekUnroll.cs#L62-L68).  Clearly it should be lower here. Probably best to un-const it and set it in cctor or Main to 1 if `! Vector<long>. IsHardwareAccelerated`.  Also fine with me to change ifdefs in some appropriate way.

**Comment by adiaaida** [comment]
Thanks!

**Comment by JosephTremoulet** [comment]
Fixed in #8861

**Comment by jkotas** [comment]
This test timed out in https://github.com/dotnet/coreclr/pull/9244

`
cmdLine:D:\\j\\workspace\\x64_release_w---0575cb46\\bin\\tests\\Windows_NT.x64.Release\\JIT\\Performance\\CodeQuality\\SIMD\\SeekUnroll\\SeekUnroll\\SeekUnroll.cmd Timed Out\r\n\n\nReturn code: -100
`
